### PR TITLE
Remove setting after uninstalling plugin

### LINKED ISSUE #265: PluginsManager does not clean up user settings after plugin removal
**Describe the bug/issue**
PluginsManager does not clean up user settings after plugin removal. Setting which contains custom hotkey is still retained after a plugin is removed.

**To Reproduce**
1. Install any plugin
2. Change the installed plugin hotkey
3. Uninstall the plugin
4. Reinstall and should see the default hotkey for the plugin, but the previously changed hotkey is still there.

This is due to the hotkey setting lives in actual Settings.json file in UserData\Settings folder. 

Have not checked if the plugin setting (in UserData folder\Settings\Plugins) is deleted or not.

### PR DESCRIPTION
**Pull request by taooceros**
fix #265 

TODO: 
create an interface for cleaning up plugin information if they store setting or some data.

### CODE REVIEW

File: Plugins/Flow.Launcher.Plugin.PluginsManager/PluginsManager.cs [hunk 1]
```diff
@@ -400,6 +402,12 @@ internal List<Result> RequestUninstall(string search)
 
         private void Uninstall(PluginMetadata plugin)
         {
+            
+
```
Comment (jjw24) [inline Plugins/Flow.Launcher.Plugin.PluginsManager/PluginsManager.cs:406]: Spaces

File: Plugins/Flow.Launcher.Plugin.PluginsManager/PluginsManager.cs [hunk 2]
```diff
@@ -400,6 +402,12 @@ internal List<Result> RequestUninstall(string search)
...
+            Core.Plugin.PluginManager.Settings.Plugins.Remove(plugin.ID);
+            Core.Plugin.PluginManager.AllPlugins.RemoveAll(p => p.Metadata.ID == plugin.ID);
```
Comment (jjw24) [inline Plugins/Flow.Launcher.Plugin.PluginsManager/PluginsManager.cs:408]: if we can do this, does this mean no restart is required?

**Comment by jjw24** [comment]
This is ready for review?

File: Plugins/Flow.Launcher.Plugin.PluginsManager/PluginsManager.cs (hunk 2 above)
Comment (taooceros) [reply Plugins/Flow.Launcher.Plugin.PluginsManager/PluginsManager.cs:408]: Well, we still cannot unload assembly unless we use the library @JohnTheGr8 has mentioned, but we can load assembly dynamically, which means that installation may not require restart.

**Comment by taooceros** [comment]
> This is ready for review?

yes, it is a simple change and it works

**Review by jjw24 — APPROVED** [review]

**Pull request merged**